### PR TITLE
Undeploy verticles backed by contextual instances

### PR DESCRIPTION
**Pull request by mkouba**
...of ApplicationScoped beans before the application context is destroyed

### CODE REVIEW

File: extensions/vertx/deployment/src/test/java/io/quarkus/vertx/verticles/NotDeployedVerticle.java [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+import javax.enterprise.context.ApplicationScoped;
+
+import io.smallrye.mutiny.Uni;
+import io.smallrye.mutiny.vertx.core.AbstractVerticle;
+
+@ApplicationScoped
+public class NotDeployedVerticle extends AbstractVerticle {
+
+    public Uni<Void> asyncStart() {
```
Comment (gastaldi) [inline extensions/vertx/deployment/src/test/java/io/quarkus/vertx/verticles/NotDeployedVerticle.java]:
```suggestion

    @Override
    public Uni<Void> asyncStart() {
```

File: extensions/vertx/deployment/src/test/java/io/quarkus/vertx/verticles/MyUndeployedVerticle.java [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+import javax.enterprise.context.ApplicationScoped;
+
+import io.smallrye.mutiny.Uni;
+import io.smallrye.mutiny.vertx.core.AbstractVerticle;
+
+@ApplicationScoped
+public class MyUndeployedVerticle extends AbstractVerticle {
+
+    public Uni<Void> asyncStart() {
```
Comment (gastaldi) [inline extensions/vertx/deployment/src/test/java/io/quarkus/vertx/verticles/MyUndeployedVerticle.java]:
```suggestion

    @Override
    public Uni<Void> asyncStart() {
```

**Review by gastaldi — APPROVED** [review]
LGTM

**Comment by gsmet** [comment]
@mkouba should we backport that or it's too much of a corner case?

**Comment by mkouba** [comment]
@gsmet it's a low priority imho but I'll let @cescoffier decide...

**Review by gsmet — DISMISSED** [review]
@mkouba could you squash the commits? Thanks!

**Pull request merged**